### PR TITLE
Update webdavfs to new fs standards

### PR DESCRIPTION
**Pull request by althonos**
Hi ! With this PR come several improvements:

### Implemented an Opener
As stated in #2, using filesystems is easier with openers. So now it is possible to do the following:
```python
import fs
with fs.open_fs('webdav://admin:admin@localhost:80/webdav') as my_webdav_fs:
    print("I didn't even import webdavfs to use a WebDAV filesystem !")
```

### Updated `WebDAVFile` to inherit from `io.RawIOBase`
Since inheriting from `io.IOBase` is now a requirement for objects returned by `FS.openbin`, I added the required code to `WebDAVFile`, and remove the code that was duplicating the stub implementation (`__enter__`, `__exit__`, etc.). I also changed the default `WebDAVFile.read()` default size value from `None` to `-1` to match the `io.RawIOBase` one.

### Fixed potential bug with seeking
`WebDAVFile.seek` was quite permissive, so I tried to have it follow the `io` convention:
* seeking from end with a positive offset raises a ValueError
* seeking from start with a negative offset raises a ValueError
* seeking from current with a negative offset stops at 0 if the offset makes the cursor go beyond

### Fixed `WebDAVFile` methods not returning anything
`WebDAVFile.write` will now return the number of bytes written, and `WebDAVFile.seek` will now return the new position after seeking.

### Made tests manage docker images internally
Instead of relying on the external docker binary, I refactored / rewrote the tests to use the `docker` python module to manage starting / stopping / removing docker images. This ensures images are launched before and removed after each test case, and that adding other test images should be easier. Tests against the `zopyx.com` servers can be unskipped by setting the `WEBDAVFS_USE_SERVERS` environment variable to `True`.

### Updated Travis-CI configuration
Since no sudo was required, I changed the build type to `sudo: false`. I made the install part use `pip` instead of `setup.py develop` since the tests should be made on a plain install to mimick a real user install. I also updated the Makefile, although it is not really needed anymore.

### Added a Python-specific `.gitignore`
Because it was easier to work with it than without it.

This now builds fine with the latest `fs` release: 
[Build Status](https://travis-ci.org/althonos/webdavfs)

### CODE REVIEW

**Comment by willmcgugan** [comment]
Thanks @althonos! I'll review soon.  WebDavFS was sponsored by Andreas Jung.

@zopyx You might want to review this as well.

**Comment by zopyx** [comment]
I can review it but it may take a while...upcoming vacation...

File: setup.py [hunk 1]
```diff
@@ -26,23 +26,25 @@
 REQUIREMENTS = [
     "fs>=2.0.0",
```
Comment (willmcgugan) [inline setup.py]: This should probably be `"fs~=2.0.8"` to only get backwards incompatible releases greater than 2.0.8. I'm sticking to semver, so any backwards incompatible changes will reset `fs` to  `3.0.0`.

Comment (althonos) [inline setup.py]: Yes, it should have been at least `fs >= 2.0.7` for the opener to work, so I'll change it to `fs ~= 2.0.7`.

File: webdavfs/webdavfs.py [hunk 2]
```diff
@@ -130,21 +115,24 @@ def seek(self, pos, whence=Seek.set):
     def tell(self):
         return self.pos
 
+    def tellable(self):
```
Comment (althonos) [inline webdavfs/webdavfs.py]: Indeed ! :)

**Comment by althonos** [comment]
Oh, and by the way, would you want me to fix the extension structure so that it installs and can be used as `fs.webdavfs` instead of `webdavfs`, as it is listed on PyPI ? I can work that out quite easily.

**Comment by willmcgugan** [comment]
Hi @zopyx Bump...

**Pull request merged**